core/fetcher/fetcher.go

```diff
@@ -146,6 +147,19 @@ func (f *Fetcher) fetchAttesterData(ctx context.Context, slot int64, defSet core
 		}
 
 		resp[pubkey] = attData
+
+		// Store Attestation data root excluding committee index.
+		eth2AttData.Index = 0
```

Comment:
are you sure we are not modifying `attData` at the same time? There are pointers involved here... Please ensure there is a test that ensures output index != 0.

Reply:
We are not modifying attData since attData is an unsigned implementation where we don't have point to attestation data.

Reply:
Also in https://github.com/ObolNetwork/charon/blob/19dd5b72466206885880e7d1f39662610dd20dff/core/fetcher/fetcher_test.go#L83
we assert committee index to be equal to the input provided.

Reply:
thanks